bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/visitors/IdentifyAggAfterJoinVisitor.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.zetasql.toolkit.antipattern.parser.visitors;
+
+import com.google.zetasql.parser.ASTNodes;
+import com.google.zetasql.parser.ASTNodes.ASTExpression;
+import com.google.zetasql.parser.ASTNodes.ASTGroupingItem;
+import com.google.zetasql.parser.ASTNodes.ASTIdentifier;
+import com.google.zetasql.parser.ASTNodes.ASTJoin;
+import com.google.zetasql.parser.ASTNodes.ASTOnClause;
+import com.google.zetasql.parser.ASTNodes.ASTOnOrUsingClauseList;
+import com.google.zetasql.parser.ASTNodes.ASTPathExpression;
+import com.google.zetasql.parser.ASTNodes.ASTSelect;
+import com.google.zetasql.parser.ASTNodes.ASTTablePathExpression;
+import com.google.zetasql.parser.ParseTreeVisitor;
+import com.google.zetasql.toolkit.antipattern.util.ZetaSQLStringParsingHelper;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+public class IdentifyAggAfterJoinVisitor extends ParseTreeVisitor {
+
+  private final String SUGGESTION_MESSAGE =
+      "GROUP By found at line number %s after Join at line number %s.Try to apply aggregation before joining to reduce join data";
```

Comment:
GROUP **BY** found at line number %s after **JOIN** at line number %s. **Consider applying** aggregation before joining to reduce join data";